yo/app/scripts/controllers/model.js

```diff
@@ -436,6 +450,59 @@ angular.module('dmpApp')
             }
         }
 
+        function openSelectRecords(project) {
+
+            var recordsSelected = true;
+
+            if (!project._$selectedRecords) {
+                project._$selectedRecords = [];
+                recordsSelected = false;
+            }
+
+            var modalInstance = $modal.open({
+                templateUrl: 'views/directives/select-records.html',
+                controller: 'SelectRecordsCtrl',
+                windowClass: 'wide',
+                resolve: {
+                    project: function() {
+                        return project;
+                    },
+                    recordsSelected: function() {
+                        return recordsSelected;
+                    }
+                }
+            });
+
+            modalInstance.result.then(function(reason) {
+
+                if(reason && (reason.removeFilter || reason.resetRecordSelection)) {
+
+                    project._$selectedRecords = null;
+                    project.selected_records = null;
```

Comment:
I dislike the usage of `null`, more in general then here in particular. I cannot track where this might lead to eventual errors later; although null behaves as false in simple if (...) checks, so it might just work. On the other hand, we differentiate at some places between `null` and `undefined`, so I would prefer to have these keys deleted instead.

```
delete project._$selectedRecords;
delete project.selected_records;
```

@chrode WDYT?